src/cargo/sources/config.rs

```diff
@@ -150,6 +156,32 @@ a lock file compatible with `{orig}` cannot be generated in this situation
             path.push(s);
             srcs.push(SourceId::for_directory(&path)?);
         }
+        if let Some(val) = table.get("git") {
+            let url = url(val, &format!("source.{}.git", name))?;
+            let try = |s: &str| {
+                let val = match table.get(s) {
+                    Some(s) => s,
+                    None => return Ok(None),
+                };
+                let key = format!("source.{}.{}", name, s);
+                val.string(&key).map(Some)
+            };
+            let reference = match try("branch")? {
+                Some(b) => GitReference::Branch(b.0.to_string()),
+                None => {
+                    match try("tag")? {
+                        Some(b) => GitReference::Tag(b.0.to_string()),
+                        None => {
+                            match try("rev")? {
+                                Some(b) => GitReference::Rev(b.0.to_string()),
+                                None => GitReference::Branch("master".to_string()),
```

Comment:
Wasn't there a bug somewhere that we shouldn't hardcode master, because it might not be the default branch?